portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +34,59 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private  List<String> messages;
+
+    @Override
+    public void init(){
+        this.messages = new ArrayList<String>();
+        this.messages.add("This site looks nice");
+        this.messages.add("I like the pictures");
+        this.messages.add("The colors are beautiful and perfect");
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Can you try moving this line into init()? e.g. `this.datastore = DatastoreServiceFactory.getDatastoreService();`. Depending on how Factory works you might be constructing a new Datastore on each request which is usually not what you want for performance reasons.

Reply:
Yes, I get you. But in the model files that was used during the  walkthroughs `DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();` was used on both the doGet() and doPost() method. Or is it meant to be otherwise ?

Reply:
I see. For consistency with the rest of the examples it's fine to leave it as is.